coalib/bearlib/aspects/Smell.py

```diff
@@ -0,0 +1,229 @@
+        importance_reason = """
+        Too large classes also known as God objects, result in ambiguous and
+        difficult to debug source code; whereas too small classes (or lazy
+        classes or freeloader) are sometimes useless.
+        """
+        fix_suggestions = """
+        Usually splitting up those classes into many other classes solves the
+        problem.
+        """
+    max_class_length = Taste[int](
+        'Represents the max number of lines for a class\'s definition.',
+        (999,), default=999)
+
+    # This shall probably be removed.
+    min_class_length = Taste[int](
+        'Represents the min number of lines for a class\'s definition.',
+        (1,), default=1)
+
+
+@ClassSmell.subaspect
+class FeatureEnvy:
+    """
+    This aspect detects occurrences of feature envy in our codebase.
+
+    `Feature envy` describes classes that excessively use methods of other
+    classes.
+    """
+    class docs:
+        example = """
+        public class Phone {
+            private final String unformattedNumber;
+            public Phone(String unformattedNumber) {
+                this.unformattedNumber = unformattedNumber;
+            }
+            public String getAreaCode() {
+                return unformattedNumber.substring(0,3);
+            }
+            public String getPrefix() {
+                return unformattedNumber.substring(3,6);
+            }
+            public String getNumber() {
+                return unformattedNumber.substring(6,10);
+            }
+        }
+        public class Customer…
+            private Phone mobilePhone;
+            public String getMobilePhoneNumber() {
+                return "(" +
+                    mobilePhone.getAreaCode() + ") " +
+                    mobilePhone.getPrefix() + "-" +
+                    mobilePhone.getNumber();
+            }
+        }
+        """
+        example_language = 'java'
+        importance_reason = """
+        This smell may occur after fields are moved to a data class; which
+        makes the code less readable, and difficult to debug.
+        """
+        fix_suggestions = """
+        Move the operations on data to the newly defined class(given that
+        the fields of one class were moved to this class) as well.
+        """
+
+
+@Smell.subaspect
+class Naming:
+    """
+    This aspect checks on identifiers in your codebase (their length
+    and the appropriate naming convention to use for them, be it variables,
+    classes or functions)
+    """
+    class docs:
+        importance_reason = """
+        Consistent use of naming convention, make the code easy to read
+        and debug.
+        """
+        fix_suggestion = """
+        Use the appropriate naming convention for each data type.
+        """
+
+    variable_naming_convention = Taste[str](
+        'Naming convention to use for variables\'s identifiers',
+        ('camelCase', 'snake_case', 'hyphenated-case', 'pascal_case'),
```

Comment:
And IMO you should better use `CamelCase` instead of `PascalCase` or even `lowerCamelCase` and `UpperCamelCase`